### DISCUSSION TITLE
Binary framework package

### DISCUSSION THREAD

**Discussion by ilendemli** [pointfreeco/swift-case-paths#143] [Ideas]
Hey!

I wanted to ask if you have considered providing precompiled swift packages for your libraries to increase build times on users and CI end. Currently swift-case-paths is depending on swift-syntax which takes quite a bit of time to compile on our CI machines.

I hope to get an answer soon.

EDIT: same with other libraries depending on this package such as swift-parsing.

**Comment by stephencelis** [comment]
@ilendemli As far as we know, Swift/SPM/Xcode does not officially support shipping binaries with macros yet. There is some discussion and exploration on the forums [here](https://forums.swift.org/t/macros-and-xcframeworks/68122), so it does appear folks could take matters into their own hands. We also know Tuist is a popular solution folks take to minimize their build times.

Given the above potential workarounds/solutions, I think we're probably most comfortable shipping the library in the vanilla way Apple seems to expect, though if you believe there are official tools that make it possible for us to do so, please let us know!
